packages/controls/src/color/color.ts

```diff
@@ -128,17 +126,36 @@ export const Color = controlTraitsRegistry.add(
     ctor.getSwatchIds = (data: ControlData | undefined): string[] =>
       data?.swatchId == null ? [] : [data.swatchId]
 
-    ctor.resolveValue = (
+    ctor.resolveValue = async (
       value: ValueType<ControlDefinition>,
       definition: ControlDefinition,
       resolver: ResourceResolver,
-    ): ResolvableValue<string | undefined> => {
+    ): Promise<string | undefined> => {
       const { swatchId, alpha } = value ?? {}
-      return resolver
-        .resolveSwatch(swatchId)
-        .map((swatch) =>
-          swatchToColorString(swatch, alpha, definition.config.defaultValue),
-        )
+      const swatch = await resolver.fetchSwatch(swatchId)
+      return swatchToColorString(swatch, alpha, definition.config.defaultValue)
+    }
+
+    ctor.subscribeValue = (
+      value: ValueType<ControlDefinition>,
+      definition: ControlDefinition,
+      resolver: ResourceResolver,
+    ): ValueSubscription<string | undefined> => {
+      const swatchSubscription = resolver.subscribeSwatch(value?.swatchId)
+
+      if (value.swatchId !== null && swatchSubscription.readValue() == null) {
+        resolver.fetchSwatch(value.swatchId).catch(console.error)
+      }
```

Comment:
Drawing attention to this: here, we make a fetch call to populate the redux store if the current value is empty. We do a similar process in `useSwatch`, with a `useEffect`. Making this fetch call gives us the expected behavior when a swatch is not in the cache: the first page render populates with the default config value, and the second render populates with the retrieved swatch.